source/base/quadrature.cc

```diff
@@ -317,6 +396,9 @@ QAnisotropic<dim>::QAnisotropic(const Quadrature<1> &qx,
         this->weights[k++] = qx.weight(k1) * qy.weight(k2);
       }
   Assert (k==this->size(), ExcInternalError());
+  this->is_tensor_product_flag = true;
+  const std::array<Quadrature<1>, 2> q_array = {qx, qy};
+  this->tensor_basis = std_cxx14::make_unique<std::array<Quadrature<1>, 2>>(q_array);
```

Comment:
you can save the temporary, I think, by just saying
```
  this->tensor_basis = std_cxx14::make_unique<std::array<Quadrature<1>, 2>>({qx, qy});
```
If that doesn't work, try using two curly braces.